src/Xamarin.Android.Build.Tasks/MSBuild/Xamarin/Android/Xamarin.Android.Resource.Designer.targets

```diff
@@ -196,6 +196,28 @@ Copyright (C) 2016 Xamarin. All rights reserved.
   </ItemGroup>
 </Target>
 
+<!--
+  The Visual Studio FastUpdate check flags the Resource Designer assembly as
+  newer than the output assembly (which it is). But it causes it to incorrectly
+  think the build is out of date. 
+
+  So lets remove it. We MUST remove the item from the @ReferencePathWithRefAssemblies
+  ItemGroup BEFORE the CollectResolvedCompilationReferencesDesignTime target runs.
+  This is because that target uses the Returns functionality. It turns out you cannot
+  modify the ItemGroup that is being returned via a Returns on a target.
+-->
+<Target Name="_RemoveResourceDesignerFromResolvedComilationReferences"
+      BeforeTargets="CollectResolvedCompilationReferencesDesignTime"
```

Comment:
We could link to this in the final commit message:
* https://github.com/dotnet/project-system/blob/dd9431b535a6158060d7ab64597276eb2c354aab/docs/design-time-builds.md#targets-that-run-during-design-time-builds
* https://github.com/dotnet/project-system/blob/dd9431b535a6158060d7ab64597276eb2c354aab/docs/up-to-date-check.md#default-inputs-and-outputs
* https://github.com/dotnet/project-system/blob/dd9431b535a6158060d7ab64597276eb2c354aab/src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/DesignTimeTargets/Microsoft.Managed.DesignTime.targets#L448-L449